src/azul/drs.py

```diff
@@ -74,10 +87,139 @@ class Access:
+        assert '/' not in self._namespace and '?' not in self._accession, self
+
+    @classmethod
+    def parse(cls, drs_uri: str) -> 'CompactDRSURI':
+        scheme, netloc = drs_uri.split('://', 1)
+        # Compact identifier-based URIs can be hard to parse when following
+        # RFC3986, with the 'namespace:accession' part matching either the
+        # heir-part or path production depending if the optional provider code
+        # and following slash is included.
+        # https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#compact-identifier-based-drs-uris
+        prefix, accession = netloc.split(':', 1)
+        reject('/' in prefix,
+               'Compact identifiers with provider codes are not supported', drs_uri)
+        reject('?' in accession,
+               'Compact identifiers must not contain query parameters', drs_uri)
+        return cls(namespace=prefix,
+                   accession=accession)
+
+    def to_url(self, client: 'DRSClient', access_id: Optional[str] = None) -> str:
+        url = client.id_client.resolve(self._namespace, self._accession)
+        require(str(url.path) == drs_object_url_path(object_id=self._accession),
+                'Unexpected DRS URL format', url)
+        url.set(path=drs_object_url_path(object_id=self._accession, access_id=access_id))
+        return str(url)
+
+
+class IdentifiersDotOrgClient:
+
+    def resolve(self, prefix: str, accession: str) -> mutable_furl:
+        namespace_id = self._prefix_to_namespace(prefix)
+        log.info('Resolved prefix %r to namespace ID %r', prefix, namespace_id)
+        resource_name, url_pattern = self._namespace_to_host(namespace_id)
+        log.info('Obtained URL pattern %r from resource %r', url_pattern, resource_name)
+        url = url_pattern.replace('{$id}', accession)
+        return furl(url)
+
+    @cached_property
+    def _http_client(self) -> urllib3.PoolManager:
+        return http_client()
+
+    _api_url = 'https://registry.api.identifiers.org/restApi/'
+
+    @cache
+    def _prefix_to_namespace(self, prefix: str) -> str:
+        prefix_info = self._api_request('namespaces/search/findByPrefix', prefix=prefix)
+        return furl(prefix_info['_links']['self']['href']).path.segments[-1]
+
+    @cache
+    def _namespace_to_host(self, namespace_id: str) -> tuple[str, str]:
+        namespace_info = self._api_request('resources/search/findAllByNamespaceId', id=namespace_id)
+        resources: MutableJSONs = namespace_info['_embedded']['resources']
+        resource = one(resources)
+        return resource['name'], resource['urlPattern']
+
+    def _api_request(self, path: str, **args) -> MutableJSON:
+        url = furl(self._api_url).add(path=path, args=args)
+        response = self._request(str(url))
+        require(response.status == 200)
+        return json.loads(response.data)
+
+    def _request(self, url: str) -> urllib3.HTTPResponse:
+        log.info('GET %s ...', url)
```

Comment:
```suggestion
        log.info('GET %s ...', url)
```

Don't we use `…` (the single ellipsis character) elsewhere?